src/Physics/babylon.physicsImpostor.ts

```diff
@@ -330,13 +330,22 @@ module BABYLON {
             }
         }
 
+        /**
+         * Legacy collision detection event support
+         */
+        public onCollideEvent: (collider:BABYLON.PhysicsImpostor, collidedWith:BABYLON.PhysicsImpostor) => void = null;
+
         //event and body object due to cannon's event-based architecture.
         public onCollide = (e: { body: any }) => {
-            if (!this._onPhysicsCollideCallbacks.length) return;

```

Comment:
My dear, I can read code :-)
There is no reason to initialize the other impostor, if no function is defined. If you ask me, it's redundant. Even if you don't ask me, it's redundant.  
I'm actually not really sure why we need this function at all, but I understand why it will be used so it's fine.

I would prefer you well check before. This function is called in each and every collision. This can happen many times in each frame, and the extra work should be avoided. I read that you see no performance impact, but if you really want I'm prove you that there is one. Trust me please, because this line of code was added for a reason.